### PR TITLE
refactor: Remove `MaxMessageSize`

### PR DESCRIPTION
**Pull request by duesee**
Just something to discuss. What do you think @jakoschiko?

### CODE REVIEW

**Comment by jakoschiko** [comment]
I don't care enough for discussing this thoroughly. If you prefer this, you can continue.

But I would prefer renaming the constructor `Fragmentizer::new_unbounded`. It's unusual to have a constructor with the name `new_*`. I would prefer `Fragmentizer::without_max_message_size`:

- It re-uses the parameter name of `Fragmentizer::new`, so it's more clear that it's the other constructor without this parameter
- It doesn't introduce another term "unbounded" that I need to understand. Does "unbounded" correlate with `max_message_size`? Or does it represent a limit in another dimension?
- The name doesn't need to be short because you should prefer `Fragmentizer::new`.

File: imap-codec/src/fragmentizer.rs [hunk 1]
```diff
@@ -131,10 +111,34 @@ pub struct Fragmentizer {
 }
 
 impl Fragmentizer {
-    pub fn new(max_message_size: MaxMessageSize) -> Self {
+    /// Create `Fragmentizer`.
```
Comment (jakoschiko) [inline imap-codec/src/fragmentizer.rs:114]:
```suggestion
    /// Creates a `Fragmentizer` with maximum message size.
```

File: imap-codec/src/fragmentizer.rs [hunk 2]
```diff
@@ -131,10 +111,34 @@ pub struct Fragmentizer {
...
+        Self {
+            unparsed_buffer: VecDeque::new(),
+            max_message_size: Some(max_message_size),
+            max_message_size_exceeded: false,
+            message_buffer: Vec::new(),
+            parser: Some(Parser::Line(LineParser::new(0))),
+        }
+    }
+
+    /// Create `Fragmentizer` without maximum message size.
```
Comment (jakoschiko) [inline imap-codec/src/fragmentizer.rs:132]:
```suggestion
    /// Creates `Fragmentizer` without maximum message size.
```

File: imap-codec/src/fragmentizer.rs [hunk 3]
```diff
@@ -131,10 +111,34 @@ pub struct Fragmentizer {
...
+    ///
+    /// <div class="warning">
```
Comment (jakoschiko) [inline imap-codec/src/fragmentizer.rs:134]: Are we ready for this level of escalation?

File: imap-codec/fuzz/fuzz_targets/fragmentizer.rs [hunk 4]
```diff
@@ -13,7 +13,10 @@ fuzz_target!(|input: (MaxMessageSize, Vec<Vec<u8>>)| {
     let mut emitted_bytes = Vec::new();
     let data_flat = data.concat();
 
-    let mut fragmentizer = Fragmentizer::new(mms);
+    let mut fragmentizer = match mms {
+        Some(max_message_size) => Fragmentizer::new(max_message_size),
+        None => Fragmentizer::without_max_message_size(),
+    };
```
Comment (jakoschiko) [inline imap-codec/fuzz/fuzz_targets/fragmentizer.rs:19]:
```suggestion
    let mut fragmentizer =
        mms.map_or_else(Fragmentizer::without_max_message_size, Fragmentizer::new);
```

Comment (jakoschiko) [reply imap-codec/fuzz/fuzz_targets/fragmentizer.rs:19]: Because it looks smarter.

File: imap-codec/src/fragmentizer.rs (hunk 3 above)
Comment (duesee) [reply imap-codec/src/fragmentizer.rs:134]: `imap-codec` already has some warnings. They are yellow and cute. No red involved.

**Comment by duesee** [comment]
You are lucky the CI takes so long... I already activated auto-merge because you said you don't care :D Thanks for the review :-)

**Comment by jakoschiko** [comment]
I love the adrenaline when reviewing against the CI.

**Pull request merged**